examples/plot_neighbors_scalar_regression.py

```diff
@@ -0,0 +1,189 @@
+"""
+Neighbors Scalar Regression
+===========================
+
+Shows the usage of the nearest neighbors regressor with scalar response.
+"""
+
+# Author: Pablo Marcos Manchón
+# License: MIT
+
+# sphinx_gallery_thumbnail_number = 3
+
+import skfda
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.model_selection import train_test_split, GridSearchCV, KFold
+from skfda.ml.regression import KNeighborsScalarRegressor
+from skfda.misc.metrics import norm_lp
+
+
+################################################################################
+#
+# In this example, we are going to show the usage of the nearest neighbors
+# regressors with scalar response. There is available a K-nn version,
+# :class:`KNeighborsScalarRegressor
+# <skfda.ml.regression.KNeighborsScalarRegressor>`, and other one based in the
+# radius, :class:`RadiusNeighborsScalarRegressor
+# <skfda.ml.regression.RadiusNeighborsScalarRegressor>`.
+#
+# Firstly we will fetch a dataset to show the basic usage.
+#
+# The caniadian weather dataset contains the daily temperature and precipitation
+# at 35 different locations in Canada averaged over 1960 to 1994.
+#
+# The following figure shows the different temperature curves.
+#
+
+data = skfda.datasets.fetch_weather()
+fd = data['data']
+
+# TODO: Change this after merge operations-with-images
+fd.axes_labels = None
+X = fd.copy(data_matrix=fd.data_matrix[..., 0])
+
+
+X.plot()
+
+
+################################################################################
+#
+# In this example we are not interested in the precipitation curves directly,
+# as in the case with regression response, we will train a nearest neighbor
+# regressor to predict a scalar magnitude.
+#
+# In the next figure the precipitation curves are shown.
+#
+
+y_func = fd.copy(data_matrix=fd.data_matrix[..., 1])
+
+plt.figure()
+y_func.plot()
+
+################################################################################
+#
+# We will try to predict the total log precipitation, i.e,
+# :math:`logPrecTot_i = \log \int_0^{365} prec_i(t)dt` using the temperature
+# curves.
+#
+# To obtain the precTot we will calculate the :math:`\mathbb{L}^1` norm of
```

Comment:
It is the integral. Sure, the value is the same, but it seems more complicated this way, I think.

Reply:
Changed, it was to avoid dealing directly with the data_matrix. A method to make definite integration may be right.